src/Apps/Search/FilterContext.tsx

```diff
@@ -0,0 +1,249 @@
+import qs from "qs"
+import React, { FC, ReactNode, useContext, useEffect, useReducer } from "react"
+import { useDidMount } from "Utils/Hooks/useDidMount"
+import { filterReducer } from "./filterReducer"
+
+export interface FilterContextValues {
+  setFilter: (name: string, value: any) => void
+  unsetFilter: (name: string) => void
+  hasFilters: boolean
+  filters: Filters
+  resetFilters: () => void
+  isDefaultValue: (name: string) => boolean
+  rangeToTuple: (name: string) => [number, number]
+}
+
+interface FilterContextProps {
+  children: ReactNode
```

Comment:
The repetition of this shape 5 times in this file irritates me, and is something I'd plan to address.

Reply:
Cool, yeah this looks like doing something as simple as 
```ts
FilterContextProps extends Filters {
  children: ReactNode
}
``` 
or
```ts
FC<Filters & { children: ReactNode }> { ... }
```
would be sufficient.